sdk/log/record.go

```diff
@@ -22,6 +23,10 @@ import (
 // cover 95% of all use-cases (https://go.dev/blog/slog#performance).
 const attributesInlineCount = 5
 
+var logAttrDropped = sync.OnceFunc(func() {
```

Comment:
I definitely get the need to avoid logging every time we drop log attributes.
But in this case, if we drop an attribute, we won't log again for the entire lifecycle of the application, which can be multiple days/months.
It feels like we should log again those dropped attributes if we haven't done so recently.

Reply:
I'm open to suggestions. How can we do that without adding allocations to the `Record` or restricting copying?

Reply:
My one idea is to log this again after some interval (e.g. 1 minute).

```go
var (
	mu   sync.Mutex
	prev time.Time
)

func BenchmarkAbc(b *testing.B) {
	b.ReportAllocs()
	b.RunParallel(func(pb *testing.PB) {
		for pb.Next() {
			now := time.Now()

			mu.Lock()
			if now.Sub(prev) > time.Microsecond { // this can be time.Minute in production
				prev = now
			}
			mu.Unlock()
		}
	})
}
```

```
cpu: Intel(R) Core(TM) i9-10885H CPU @ 2.40GHz
BenchmarkAbc-16    	 5651463	       211.5 ns/op	       0 B/op	       0 allocs/op
PASS
```

However, I do not like setting some arbitrary global value which the user would also be not able to control. 

The second idea would be reporting via `otel.Handle` and leaving the logging decision to the user. However, then the default behavior would break:

> To prevent excessive logging, the message MUST be printed at most once per LogRecord (i.e., not per discarded attribute).

Personally, I am fine with the current implementation.

Reply:
> `sync.Once` uses an `atomic.Uint32` which will default to 0, and it sets to 1 when the execution happens. We could reimplement something similar which would increment the atomic integer, and log if `number % 1000` for example. It definitely adds complexity, but should be doable with no allocations.
> 
> I'm also happy to get this merged as-is, and look into it in another PR.

The `sync.Mutex` is important in that implementation. In order to not double count it is needed. Adding a mutex means adding a "no copy" requirement on the record.

Reply:
> Adding a mutex means adding a "no copy" requirement on the record.

I think such requirement is fine. User should use Clone to copy a record

Reply:
Hmm, our algorithm doesn't need to also grantee "completion on return" so we could just use a CAS approach.

I'll look at updating.

Reply:
`atomic.Uint32` has the same "no copy" restriction:

https://cs.opensource.google/go/go/+/refs/tags/go1.22.3:src/sync/atomic/type.go;l=116

We will again need to add that restriction to the `Record` or an allocation to track a referenceable value.

Reply:
I am fine with the current implementation.